waku/v2/node/rest/debug_api.nim

```diff
@@ -0,0 +1,94 @@
+import
+  stew/byteutils,
+  chronicles,
+  json_serialization,
+  json_serialization/std/options,
+  presto/[route, client]
+import "."/[serdes, utils],
+  ../wakunode2
+
+logScope: topics = "rest_api_debug"
+
+
+#### Types
+
+type
+  DebugWakuInfo* = object
+    listenAddresses*: seq[string]
+    enrUri*: Option[string]
+
+
+#### Serialization and deserialization
+
+proc writeValue*(writer: var JsonWriter[RestJson], value: DebugWakuInfo)
+  {.raises: [IOError, Defect].} =
+  writer.beginRecord()
+  writer.writeField("listenAddresses", value.listenAddresses)
+  if value.enrUri.isSome:
+    writer.writeField("enrUri", value.enrUri)
+  writer.endRecord()
+
+proc readValue*(reader: var JsonReader[RestJson], value: var DebugWakuInfo)
+  {.raises: [SerializationError, IOError, Defect].} =
+  var
+    listenAddresses: Option[seq[string]]
+    enrUri: Option[string]
+
+  for fieldName in readObjectFields(reader):
+    case fieldName
+    of "listenAddresses":
+      if listenAddresses.isSome():
+        reader.raiseUnexpectedField("Multiple `listenAddresses` fields found", "DebugWakuInfo")
+      listenAddresses = some(reader.readValue(seq[string]))
+    of "enrUri":
+      if enrUri.isSome():
+        reader.raiseUnexpectedField("Multiple `enrUri` fields found", "DebugWakuInfo")
+      enrUri = some(reader.readValue(string))
+    else:
+      unrecognizedFieldWarning()
+
+  if listenAddresses.isNone():
+    reader.raiseUnexpectedValue("Field `listenAddresses` is missing")
+
+  value = DebugWakuInfo(
+    listenAddresses: listenAddresses.get,
+    enrUri: enrUri
+  )
+
+
+####  Server request handlers
+
+proc toDebugWakuInfo(nodeInfo: WakuInfo): DebugWakuInfo = 
+    DebugWakuInfo(
+      listenAddresses: nodeInfo.listenAddresses,
+      enrUri: some(nodeInfo.enrUri)
+    )
+
+const ROUTE_DEBUG_INFOV1* = "/debug/v1/info"
```

Comment:
If we believe maintenance on this constant may be necessary in future, can we move it top level (before the `type`s)?

Reply:
I am curious about the maintenance work you foresee with this constant.

The idea is to have a constant per path. So the string literal is not within the handler code. So for example, if we come up with an "info V2" endpoint then we would have another constant collocated to the "info V2" route handler method:

```nim
const ROUTE_DEBUG_INFOV2* = "/debug/v2/info"
```